contracts/erc721/UserMintableERC721.sol

```diff
@@ -430,6 +464,65 @@ contract UserMintableERC721 is Initializable, ERC721, ERC721Enumerable, ERC721Me
     );
   }
 
+  /**
+   * @dev Checks that provided document is signed by an identity
+   * and validates and checks if the public key used is a valid
+   * SIGNING_KEY
+   * @param documentRoot bytes32 the anchored document root
+   * @param identity address Identity that signed the document
+   * @param signingRoot bytes32 The message that was signed
+   * @param signature bytes The signature
+   * used to contract the property for precise proofs
+   * @param salt bytes32 salt for leaf construction
+   * @param proof bytes32[] proofs for leaf construction
+   */
+
+  function _requireSignedByIdentity(
```

Comment:
We need a merkle proof that the identityAddress is listed as a collaborator otherwise I could just add signatures which are actually no collaborator on the document.

Reply:
This is a helper method. It is intended to me more generic. In the PO contract we send the msg.sender as a param and validate that is a sender on the document with another method. This translates to: The sender of the invoice signed the document